src/alarms/alarm-handling.md

```diff
@@ -0,0 +1,87 @@
+
+# Introduction
+That document specifies alarm handling for thin-edge.
+
+# Data Semantic of Alarms

```

Comment:
I fully agree on the statement that alarms are related to an observable state. An alarm is *on* or *off*.

However, I think the difference between a state and a stream of state-transitions is deeper than just avoiding message losses. One can have a reliable stream of state-transitions and rebuild the current state from that stream - this how Cumulocity implements alarm support. With an observable state - that can be requested on-demand, one doesn't need to follow the transitions, one can just ask for the latest value and more importantly, in the context of MQTT, one can receive this latest value as often as requested not just on a state transition.

This is why using the *retain* flag is appropriate for alarms. If this flag is set on publish for all alarms, then:
* A subscriber can follow the state-transitions as usual and this stream reliability can be controlled with the QoS of the subscriber.
* On each (re)-connection the subscriber receives only the current state of the alarms - there is no point to be warned for alarms that have been cleared meantime.
* Even if there is no alarm state changes, a subscriber receives on-connect the current state of the alarms.

Reply:
I agree, good point! I will add according detail to the spec.